File: python/examples/exponential_concatenation.py
#!/usr/bin/python
# writen by andrewt@cse.unsw.edu.au as a COMP2041 example
# create a string of size 2^n by concatenation
import sys
if len(sys.argv) != 2:
	print >>sys.stderr, "Usage: %s <n>" % sys.argv[0]
	sys.exit(1)
n = 0
string = '@'
while  n  < int(sys.argv[1]):
    string =  string + string
    n += 1
print "String of 2^%d = %d characters created\n" % (n, len(string));

File: perl/examples/exponential_concatenation.pl
#!/usr/bin/perl -w
# writen by andrewt@cse.unsw.edu.au as a COMP2041 example
# create a string of size 2^n by concatenation

die "Usage: $0 <n>\n" if @ARGV != 1;
$n = 0;
$string = '@';
while ($n  < $ARGV[0]) {
    $string = "$string$string";
    $n++;
}
printf "String of 2^%d = %d characters created\n", $n, length $string;
